tests/Unit/CrudPanel/CrudPanelColumnsTest.php

```diff
@@ -110,6 +112,112 @@ class CrudPanelColumnsTest extends BaseDBCrudPanelTest
         ],
     ];
 
+    private $expectedRelationColumnsArray = [
+        'accountDetails' => [
+            'name'        => 'accountDetails',
+            'label'       => 'AccountDetails',
+            'type'        => 'relationship',
+            'key'         => 'accountDetails',
+            'priority'    => 0,
+            'tableColumn' => false,
+            'orderable'   => false,
+            'searchLogic' => false,
+            'entity'      => 'accountDetails',
+            'model'       => 'Backpack\CRUD\Tests\Unit\Models\AccountDetails',
+            'relation_type' => 'HasOne',
+        ],
+        'accountDetails__nickname' => [
+            'name'        => 'accountDetails.nickname',
+            'label'       => 'AccountDetails.nickname',
+            'type'        => 'relationship',
+            'key'         => 'accountDetails__nickname',
+            'priority'    => 1,
+            'tableColumn' => false,
+            'orderable'   => false,
+            'searchLogic' => false,
+            'relation_type' => 'HasOne',
+            'entity' => 'accountDetails.nickname',
```

Comment:
Hmmm I fear this entity string already shows the problem mentioned above: in the [select view](https://github.com/Laravel-Backpack/CRUD/blob/master/src/resources/views/crud/columns/select.blade.php#L9) this entity is passed as nested relation, so that the crudpanel attempts to call the `nickname` method on `accountDetails` which doesn't exist.

There probably needs to be some better way to distinguish between attribute or relation and the end of a 'dotted string' and it might be wise to add some unit tests calling something similar to the call `$crud->getRelatedEntriesAttributes($entry, $column['entity'], $column['attribute']);` that is used in the `select` view.